src/clj/tasks/load_generator.clj

```diff
@@ -0,0 +1,141 @@
+                            :client-id runner-client-id
+                            :?data {:gameid game-id
+                                    :password ""}})
+      ;; Select decks
+      (game-ws-handler/-msg-handler {:id :lobby/deck :ring-req {:user {:username "sample2"}}
+                            :client-id corp-client-id
+                            :?data (str (some #(if (= "Corp" ((% :identity) :side)) (% :_id) false) corp-decks))}) ;; find one deck where :identity :side "Corp", then get the _id
+      (game-ws-handler/-msg-handler {:id :lobby/deck :ring-req {:user {:username "sample3"}}
+                            :client-id runner-client-id
+                            :?data (str (some #(if (= "Runner" ((% :identity) :side)) (% :_id) false) runner-decks))}) ;; find one deck where :identity :side "Runner", then get the _id
+      (doall
+        (pmap
+          (fn [n]
+            (let [userClientID (uuid/to-string (uuid/v1))
+                  creds (login (str "sample" n) "password")
+                  socket (ws/connect (str "ws://localhost:1042/ws?client-id=" userClientID "&csrf-token=" (URLEncoder/encode (creds "X-CSRF-Token")))
+                                     :client client
+                                     :on-error #(safe-println "spectator error" %)
+                                     :on-close #(safe-println 'closed %1 %2)
+                                     :headers creds)]
+              (game-ws-handler/-msg-handler {:id :lobby/watch :ring-req {:user {:username (str "sample" (+ 4 n))}}
+                                    :client-id userClientID
+                                    :?data {:gameid game-id
+                                            :password ""}})
+              socket))
+          (range maxUsers)))
+
+      (safe-println "Spectators connected")
+      (game-ws-handler/-msg-handler {:id :netrunner/start :ring-req {:user {:username "sample2"}}
+                          :client-id corp-client-id})
+      (safe-println "Started game"))
+      ))
+
+(defn usage
+  [options-summary]
+  (->> [""
+        "Usage: lein load-generator [options]"
+        ""
+        "Options:"
+        options-summary]
+       (join \newline)))
+
+(def cli-options
+   [["-n" "--num NUM" "Number of spectators to connect to game"
+    :parse-fn #(Integer/parseInt %)
+    :validate [#(< 0 % 0x10000) "Must be a number between 0 and 65536"]
+    :default 1000]])
+
+(defn exit [status msg]
+  (binding [*out* *err*]
+    (println msg))
+  (System/exit status))
+
+(defn command
+  "This is not really a load test per se, but a development tool to assist debugging performance issues.
+
+  This will start the server and create a single game on your environment between users sample2 (Corp) and sample3 (Runner).
+  One thousand spectators will be connected to the game by default, from users sample4-sample1003.
+
+  The load generator must be run AFTER creating sample users."
```

Comment:
Do the sample users have to be manually created?

Reply:
and if they do, should you include a method for automatically creating those users, maybe behind a flag in this task?